login.go

```diff
@@ -39,69 +42,116 @@ func (cmd *loginCommand) Register(fs *flag.FlagSet) {
-		logrus.Warnf("WARNING! Using --password via the CLI is insecure. Use --password-stdin.")
+	var dcfg *configfile.ConfigFile
+	var authConfig types.AuthConfig
+	var err error
+
+	if !cmd.authConfig {
+		if cmd.password != "" {
+			logrus.Warnf("WARNING! Using --password via the CLI is insecure. Use --password-stdin.")
+			if cmd.passwordStdin {
+				return errors.New("--password and --password-stdin are mutually exclusive")
+			}
+		}
+
+		// Handle when the password is coming over stdin.
 		if cmd.passwordStdin {
-			return errors.New("--password and --password-stdin are mutually exclusive")
+			if cmd.user == "" {
+				return errors.New("must provide --username with --password-stdin")
+			}
+
+			// Read from stadin.
+			contents, err := ioutil.ReadAll(os.Stdin)
+			if err != nil {
+				return err
+			}
+
+			cmd.password = strinpEndlines(string(contents))
 		}
-	}
 
-	// Handle when the password is coming over stdin.
-	if cmd.passwordStdin {
-		if cmd.user == "" {
-			return errors.New("must provide --username with --password-stdin")
+		if len(args) > 0 {
+			cmd.serverAddress = args[0]
 		}
 
+		// Set the default registry server address.
+		if cmd.serverAddress == "" {
+			cmd.serverAddress = defaultDockerRegistry
+		}
+
+		// Get the auth config.
+		dcfg, authConfig, err = configureAuth(cmd.user, cmd.password, cmd.serverAddress)
+		if err != nil {
+			return err
+		}
+	} else {
+		dcfg, err = config.Load(config.Dir())
 		// Read from stadin.
 		contents, err := ioutil.ReadAll(os.Stdin)
 		if err != nil {
 			return err
 		}
 
-		cmd.password = strings.TrimSuffix(string(contents), "\n")
-		cmd.password = strings.TrimSuffix(cmd.password, "\r")
-	}
-
-	if len(args) > 0 {
-		cmd.serverAddress = args[0]
-	}
+		authConfigStr := strinpEndlines(string(contents))
 
-	// Set the default registry server address.
-	if cmd.serverAddress == "" {
-		cmd.serverAddress = defaultDockerRegistry
-	}
-
-	// Get the auth config.
-	dcfg, authConfig, err := configureAuth(cmd.user, cmd.password, cmd.serverAddress)
-	if err != nil {
-		return err
+		if authConfig, err = base64DecodeAuth(authConfigStr); err != nil {
+			return err
+		}
 	}
 
 	// Attempt to login to the registry.
 	r, err := registryapi.New(cliconfigtypes2dockerapitypes(authConfig), registryapi.Opt{Debug: debug})
 	if err != nil {
 		return fmt.Errorf("creating registry client failed: %v", err)
 	}
-	token, err := r.Token(r.URL)
-	if err != nil && err != registryapi.ErrBasicAuth {
-		return fmt.Errorf("getting registry token failed: %v", err)
-	}
 
-	// Configure the token.
-	if token != "" {
-		authConfig.Password = ""
-		authConfig.IdentityToken = token
+	if authConfig.IdentityToken != "" {
+		token, err := r.Token(r.URL)
+		if err != nil && err != registryapi.ErrBasicAuth {
+			return fmt.Errorf("getting registry token failed: %v", err)
+		}
+
+		// Configure the token.
+		if token != "" {
+			authConfig.Password = ""
+			authConfig.IdentityToken = token
+		}
 	}
 
 	// Save the config value.
```

Comment:
I think this can be removed as we are using it inside a container and don't wish to create side-effect

Reply:
This is the login process. The pull/push process needs the saved credentials in order to authenticate to the remote registry.

Not saving the credentials will result in an error when trying to pull images.